compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTableTypeSymbol.java

```diff
@@ -70,16 +70,8 @@ public Optional<TypeSymbol> keyConstraintTypeParameter() {
 
     @Override
     public List<String> keySpecifiers() {
-        if (this.keySpecifiers == null) {
-            List<String> specifiers = ((BTableType) this.getBType()).fieldNameList;
-
-            if (specifiers == null) {
-                specifiers = new ArrayList<>();
-            }
-
-            this.keySpecifiers = Collections.unmodifiableList(specifiers);
-        }
-
+        List<String> specifiers = ((BTableType) this.getBType()).fieldNameList;
```

Comment:
Why should we always do this instead of passing the module level var?

Reply:
It seems we have populated the `keySpecifiers`(module-level var) in this function itself.

Reply:
That's fine; my point was it should be populated just once. ie; when `this.keySpecifier` is empty

Reply:
got it! yes, that's correct. will fix.